markdownlint/docs/TOP002.md

```diff
@@ -0,0 +1,21 @@
+# `TOP002` - No Inline Code in Headings
+
+Tags: `headings`
+
+Aliases: `no-code-headings`
+
+This rule is triggered when a heading contains inline code. For example:
+
+```markdown
+## Using `console.log()` for Debugging
+```
+
+To fix this issue, remove the inline code from the heading and rephrase it to convey the same meaning without using code:
+
+```markdown
+## Using console.log for Debugging
+```
+
+## Rationale
+
+Headings should be concise and descriptive, providing a clear overview of the section's content. Including inline code in headings can make them less readable and may not effectively convey the purpose of the section to users.
```

Comment:
@KevinMulhern do you recall the reason we started omitting inline code in headings? I thought it was due to how the ID's are created on the heading elements not playing well with backticks, but can't actually recall.

Reply:
@cakegod so it seems like the reason we may have stopped using inline code in headings was (at least partially) due to how the ID and anchor elements were rendered. The backticks/inline code would cause things to render a little wonky (duplication of the heading title).

I think what you have here is fine for now, though, as it helps prevent someone from wanting to make a heading that's verbose code (rather than something that would be fine, such as wrapping specific keywords in backticks e.g. "The `lang` attribute").

Reply:
Should we add that info to the rationale?

Reply:
We can do that in a followup. It may not be an issue any longer, but requires a little more investigating that I wouldn't want holding this PR up for.